### PR TITLE
Feature/6723 upgrade elasticsearch

### PR DESCRIPTION
**Pull request by melsawy**

### CODE REVIEW

File: config/initializers/location_search.rb [hunk 1]
```diff
@@ -23,33 +24,26 @@ def lookup_type(type)
...
     geojson = JSON.parse(self.get_field_value(:response_geolocation))
     coordinates = geojson['geometry']['coordinates']
     indexable = geojson['properties']['name']
 
     if coordinates[0] != 0 || coordinates[1] != 0
+      # re-compute long value before sending to Elasticsearch
       location = {
         lat: coordinates[0],
-        lon: coordinates[1]
+        lon: ((coordinates[1].to_f + 180) % 360) - 180
```
Comment (infojunkie) [inline config/initializers/location_search.rb]: what is this for?

File: db/migrate/20170720090144_add_es_mapping_for_account.rb [hunk 2]
```diff
@@ -1,5 +1,6 @@
 class AddEsMappingForAccount < ActiveRecord::Migration
   def change
+    return
```
Comment (infojunkie) [inline db/migrate/20170720090144_add_es_mapping_for_account.rb]: if this migration is no longer needed, we can delete it

File: lib/check_elastic_search.rb [hunk 3]
```diff
@@ -1,133 +1,144 @@
...
+    # data.merge!(add_missing_fields(options))
+    fields = {'updated_at' => Time.now.utc}
+    options[:keys].each{|k| fields[k] = data[k] if !data[k].blank? }
+    client = MediaSearch.gateway.client
+    client.update index: CheckElasticSearchModel.get_index_alias, type: 'media_search', id: options[:doc_id], retry_on_conflict: 3,
+                  body: { doc: fields }
   end
 
-  def add_update_media_search_child(child, keys, data = {}, parent = nil)
+  # Keep this method until verify search feature.
```
Comment (infojunkie) [inline lib/check_elastic_search.rb]: what's that for?

File: lib/tasks/check_create_es_data_from_pg.rake [hunk 4]
```diff
@@ -3,25 +3,75 @@ namespace :check do
...
+            doc.save!
+          rescue Exception => e
+            failed_items << {error: e, obj_id: obj.id, obj_class: obj.class.name}
+          end
+        end
+        print '.'
       end
-  	end
+    end
+    puts "Failed to index #{failed_items.size} items"
```
Comment (infojunkie) [inline lib/tasks/check_create_es_data_from_pg.rake]: If zero failures plz print nothing

File: config/initializers/location_search.rb (hunk 1 above)
Comment (melsawy) [reply config/initializers/location_search.rb]: new versions of ES validate longitude value allowed range is  [-180..180]
I got an error `illegal longitude value`  while creating index from PG and used this formula to fix this error.

File: db/migrate/20170720090144_add_es_mapping_for_account.rb (hunk 2 above)
Comment (melsawy) [reply db/migrate/20170720090144_add_es_mapping_for_account.rb]: yes not needed, will delete it

File: lib/check_elastic_search.rb (hunk 3 above)
Comment (melsawy) [reply lib/check_elastic_search.rb]: This method `add_missing_fields` was added to index all parent filed when using versions, now I use retry_on_conflict and just keep it until verify everything on QA then will remove it.
I did many tests locally and everything works well without this method.

File: lib/tasks/check_create_es_data_from_pg.rake (hunk 4 above)
Comment (melsawy) [reply lib/tasks/check_create_es_data_from_pg.rake]: Done

**Pull request merged**